app/services/payment_provider_customers/factory.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module PaymentProviderCustomers
+  class Factory
+    def self.new_instance(provider_customer:)
+      service_class(provider_customer).new(provider_customer)
+    end
+
+    private
+
+    def self.service_class(provider_customer)
```

Comment:
Lint/IneffectiveAccessModifier: `private` (on line 9) does not make singleton methods private. Use `private_class_method` or `private` inside a `class << self` block instead.